src/js/components/accordion.js

```diff
@@ -1,122 +1,76 @@
+ * Toggle a button's "pressed" state, optionally providing a target
+ * state.
  *
- * @param {HTMLElement} el An HTMLElement to turn into an accordion.
+ * @param {HTMLButtonElement} button
+ * @param {boolean?} expanded If no state is provided, the current
+ * state will be toggled (from false to true, and vice-versa).
+ * @return {boolean} the resulting state
  */
-function Accordion (el) {
-  var self = this;
-  this.root = el;
-
-  // delegate click events on each <button>
-  this.$(SELECTOR_BUTTON).forEach(function (button) {
-    if (button.attachEvent) {
-      button.attachEvent('onclick', showPanelListener.bind(self, button));
-    } else {
-      button.addEventListener('click', showPanelListener.bind(self, button));
-    }
-  });
-
-  // find the first expanded button
-  var expanded = this.select(SELECTOR_BUTTON_EXPANDED);
-  this.hideAll();
-  if (expanded !== undefined) {
-    this.show(expanded);
+const toggleButton = (button, expanded) => {
+  var accordion = button.closest(ACCORDION);
+  if (!accordion) {
+    throw new Error(`${BUTTON} is missing outer ${ACCORDION}`);
   }
-}
 
-/**
- * @param {String} selector
- * @return {Element}
- */
-Accordion.prototype.select = function (selector) {
-  return this.$(selector)[ 0 ];
-};
+  expanded = toggle(button, expanded);
+  // XXX multiselectable is opt-in, to preserve legacy behavior
+  const multiselectable = accordion.getAttribute(MULTISELECTABLE) === 'true';
 
-/**
- * @param {String} selector
- * @return {Array.HTMLElement}
- */
-Accordion.prototype.$ = function (selector) {
-  return select(selector, this.root);
-};
-
-/**
- * @param {HTMLElement} button
- * @param {Boolean} expanded
- */
-Accordion.prototype.toggle = function (button, expanded) {
-  var target = getTargetOf(button);
-  button.setAttribute('aria-expanded', expanded);
-  target.setAttribute('aria-hidden', !expanded);
-  return this;
-};
-
-/**
- * @param {HTMLElement} button
- * @return {Accordion}
- */
-Accordion.prototype.hide = function (button) {
-  return this.toggle(button, false);
-};
-
-/**
- * @param {HTMLElement} button
- * @return {Accordion}
- */
-Accordion.prototype.show = function (button) {
-  return this.toggle(button, true);
+  if (expanded && !multiselectable) {
+    forEach(getAccordionButtons(accordion), other => {
+      if (other !== button) {
+        toggle(other, false);
+      }
+    });
+  }
 };
 
 /**
- * @return {Accordion}
+ * Get an Array of button elements belonging directly to the given
+ * accordion element.
+ * @param {HTMLElement} accordion
+ * @return {array<HTMLButtonElement>}
  */
-Accordion.prototype.hideAll = function () {
-  var self = this;
-  this.$(SELECTOR_BUTTON).forEach(function (button) {
-    self.hide(button);
+const getAccordionButtons = accordion => {
+  return filter(select(BUTTON, accordion), button => {
+    return button.closest(ACCORDION) === accordion;
   });
-  return this;
 };
 
-module.exports = Accordion;
+module.exports = behavior({
```

Comment:
Along the same vein of using `import`, consider using `export`/`export default`. Perhaps in a separate PR.